### ISSUE TITLE
Program Drawer isn't screen reader or keyboard accessible 

### ISSUE THREAD

**Issue by pdpinch** [mitodl/mitxonline#904]
### Steps to Reproduce

1. Visit the dashboard with a screenreader like NVDA or Apple VoiceOver

### Expected Behavior

1. focus should shift to the drawer when it is opened 
2. can use keyboard to tab through links in drawer when it is open 

### Actual Behavior

1. Screenreader does not have an alert to know to jump to menu. Screen reader users have no idea the menu is there.

### Related 

Can we implement the drawer similar to a "fly-out menu", https://www.w3.org/WAI/tutorials/menus/flyout/ ?
